packages/ffmpeg.rb

```diff
@@ -1,121 +1,217 @@
-           '--enable-gnutls',
-           '--enable-libass',
-           "--arch=#{ARCH}",
-           '--enable-gray',
-           '--enable-gpl'
+    case ARCH
+    when 'i686'
+      @mfx = '--enable-libmfx'
+      @lto = ''
+      @enablelto = ''
+    when 'x86_64'
+      @mfx = '--enable-libmfx'
+      @lto = '-flto'
+      @enablelto = '--enable-lto'
+    when 'aarch64', 'armv7l'
+      @mfx = ''
+      @lto = '-flto'
+      @enablelto = '--enable-lto'
+    end
+
+    @git_dir = 'ffmpeg_git'
+    @git_hash = "n#{@_ver}"
+    @git_url = 'https://git.ffmpeg.org/ffmpeg.git'
+    FileUtils.rm_rf(@git_dir)
+    FileUtils.mkdir_p(@git_dir)
+    system "git clone #{@git_url} #{@git_dir}"
+    Dir.chdir @git_dir do
+      system "git checkout #{@git_hash}"
+      system 'git cherry-pick -n 7c59e1b0f285cd7c7b35fcd71f49c5fd52cf9315' # fix build against libsrt 1.4.2
+      # Patch from https://raw.githubusercontent.com/archlinux/svntogit-packages/packages/ffmpeg/trunk/vmaf-model-path.patch
+      @vmaf_patch = <<~'VMAF_PATCH_EOF'
+        diff --git a/libavfilter/vf_libvmaf.c b/libavfilter/vf_libvmaf.c
+        index 249e50c720..9b791e19b1 100644
+        --- a/libavfilter/vf_libvmaf.c
+        +++ b/libavfilter/vf_libvmaf.c
+        @@ -72,7 +72,7 @@ typedef struct LIBVMAFContext {
+         #define FLAGS AV_OPT_FLAG_FILTERING_PARAM|AV_OPT_FLAG_VIDEO_PARAM
+         
+         static const AVOption libvmaf_options[] = {
+        -    {"model_path",  "Set the model to be used for computing vmaf.",                     OFFSET(model_path), AV_OPT_TYPE_STRING, {.str="/usr/local/share/model/vmaf_v0.6.1.pkl"}, 0, 1, FLAGS},
+        +    {"model_path",  "Set the model to be used for computing vmaf.",                     OFFSET(model_path), AV_OPT_TYPE_STRING, {.str="/usr/share/model/vmaf_v0.6.1.pkl"}, 0, 1, FLAGS},
+             {"log_path",  "Set the file path to be used to store logs.",                        OFFSET(log_path), AV_OPT_TYPE_STRING, {.str=NULL}, 0, 1, FLAGS},
+             {"log_fmt",  "Set the format of the log (xml or json).",                            OFFSET(log_fmt), AV_OPT_TYPE_STRING, {.str=NULL}, 0, 1, FLAGS},
+             {"enable_transform",  "Enables transform for computing vmaf.",                      OFFSET(enable_transform), AV_OPT_TYPE_BOOL, {.i64=0}, 0, 1, FLAGS},
+      VMAF_PATCH_EOF
+      IO.write('vmaf.patch', @vmaf_patch)
+      system 'patch -p1 -i vmaf.patch'
+      # ChromeOS awk employs sandbox redirection protections which screw
+      # up configure script generation, so use mawk.
+      system "sed -i 's/awk/mawk/g' configure"
+      system "env CFLAGS='-pipe -fno-stack-protector -U_FORTIFY_SOURCE #{@lto} -fuse-ld=gold' \
+        CXXFLAGS='-pipe -fno-stack-protector -U_FORTIFY_SOURCE #{@lto} -fuse-ld=gold' \
+        LDFLAGS='-fno-stack-protector -U_FORTIFY_SOURCE #{@lto}' \
+        CC=clang CXX=clang++ \
+        ./configure \
+        --arch=#{ARCH} \
+        #{CREW_OPTIONS.sub(/--build=.*/, '')} \
```

Comment:
Does `--build=.*` remove anything else other than the build option?

Reply:
testing `#{CREW_OPTIONS.sub(/--build=(.*?)\s/, '')}`

Reply:
Doesn't work. Maybe there's some special common autoconf code which recognizes  build, target, host, etc, or none of them. So this seems the best way to deal with it,

Reply:
Try to replace `\s` to space?

Reply:
What is the output of `#{CREW_OPTIONS.sub(/--build=.*/, //)}`?

Reply:
The variables after `--build` kill the build, which is why the regex used turned out to be ok.